salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,190 @@
+                )
+            )
+
+    @parameterized.expand([
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES["get_kubeconfig"]
+    ])
+    def test_get_kubeconfig(
+        self,
+        api_server_pillar,
+        config_options,
+        kwargs,
+        result,
+    ):
+        """
+        Tests the return of `get_kubeconfig` function
+        """
+        pillar_dict = {}
+        if api_server_pillar:
+            pillar_dict = {'metalk8s': {'api_server': api_server_pillar}}
+
+        config_options_dict = {}
+        if config_options:
+            config_options_dict.update(config_options)
+
+        config_options_mock = MagicMock(side_effect=config_options_dict.get)
+
+        salt_dict = {
+            'config.option': config_options_mock,
+        }
+
+        with patch.dict(metalk8s_kubernetes_utils.__pillar__, pillar_dict), \
+                patch.dict(metalk8s_kubernetes_utils.__salt__, salt_dict):
+            self.assertEqual(
+                tuple(result),
+                metalk8s_kubernetes_utils.get_kubeconfig(**kwargs)
+            )
+
+    @parameterized.expand([
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES["get_version_info"]
+    ])
+    def test_get_version_info(
+        self,
+        kubeconfig,
+        api_client,
+        result,
+        raises=False,
+        **kwargs
+    ):
+        """
+        Tests the return of `get_version_info` function
+        """
+        kubeconfig = MagicMock(return_value=tuple(kubeconfig))
+
+        apiclient = MagicMock(return_value=api_client)
+        api_instance = MagicMock()
+
+        if api_client is None or api_instance is None:
+            api_instance.return_value.get_code.side_effect = \
+                ApiException('Failed to get version info')
+        api_instance.return_value.get_code.return_value.to_dict.return_value  \
+            = result
+
+        with patch("metalk8s_kubernetes_utils.get_kubeconfig", kubeconfig), \
+                patch("kubernetes.config.new_client_from_config", apiclient), \
+                patch("kubernetes.client.VersionApi", api_instance):
+            if raises:
+                self.assertRaisesRegexp(
+                    CommandExecutionError,
+                    result,
+                    metalk8s_kubernetes_utils.get_version_info
+                )
+            else:
+                self.assertEqual(
+                    metalk8s_kubernetes_utils.get_version_info(**kwargs),
+                    result
+                )
+
+    def test_ping(self):
+        """
+        Tests the return of `ping` function
+        """
+        version_info = {"major": "1", "go_version": "go1.13.8"}
+        version_info_mock = MagicMock()
+
+        version_info_mock.return_value.get_version_info.return_value = \
+            version_info
+
+        with patch(
+            "metalk8s_kubernetes_utils.get_version_info", version_info_mock
+        ):
```

Comment:
We do not care about the return of `get_version_info` here we just want this function to raise or not to test both case (here you test only the True case)